Add docs about cairo 1 modules (#1603)
software-mansion/protostar: website/docs/tutorials/08-cairo-1-support/01-project-initialization.md

    - Single test file with one test for function defined in `main.cairo`
- `protostar.toml` containing information about the project

## Cairo 1 modules

> In order to understand how to create cairo1 modules, we need to talk about `cairo_project.toml` and `lib.cairo` purpose.

Reviewer: ```suggestion
In order to understand how to create Cairo 1.0 modules, we need to talk about `cairo_project.toml` and `lib.cairo` purpose.
```

> In order to understand how to create Cairo 1.0 modules, we need to talk about `cairo_project.toml` and `lib.cairo` purpose.

Reviewer 2: ```suggestion
In order to understand how to create Cairo 1.0 modules, we need to talk about the purpose of `cairo_project.toml` and `lib.cairo`.
```

### Project defaults

> #### 1. `cairo_project.toml`

Reviewer: Shouldn't cairo_project be autogenerated by Protostar? That's what's happening in Scarb

Author: Yes it is - that's why it's listed under "project defaults"

Author: I mean, we don't add to this file automatically if you meant that

> It is needed for definition of crate roots, so the places where `lib.cairo` files are located

Reviewer 2: ```suggestion
It is needed for the definition of crate roots, which are the places where `lib.cairo` files are located.
```

> Content of the default `cairo_project.toml` contains only definition of `src` crate 

Reviewer 2: ```suggestion
The default `cairo_project.toml` file contains only the definition of the `src` crate 
```

```toml
[crate_roots]
src = "."
```

> The `src` crate is then imported in our tests in following manner:

Reviewer 2: ```suggestion
The `src` crate is then imported into our tests in the following manner:
```

Author: That sounds incorrect to me

```
use src::main::fib;
```

#### 2. `lib.cairo`

> It is the root of the module tree of the package. here you can define functions, declare used modules, etc.

Reviewer 2: ```suggestion
It is the root of the module tree of the package. Here you can define functions, declare used modules, etc.
```

> The default one has only `main` module declaration:

Reviewer 2: ```suggestion
The default one has only the `main` module declaration:
```

```
mod main;
```
### Creating and using a new module

> Suppose we wanted to create a module called `mod1` inside the `src` crate, and use it in tests.

Reviewer 2: ```suggestion
Suppose we wanted to create a module called `mod1` inside the `src` crate and use it in tests.
```

> Here are the steps we need to do:
> 
> 1. Create `mod1` directory in `src` crate
> 2. Create `mod1.cairo` alongside this directory. 
> 3. Create your source file inside of `mod1` (i.e. `functions.cairo` or any suitable name). Define your code here.
> 4. Declare the source file/files in `mod1.cairo`. The file contents should look like this (assuming you have `functions.cairo` from previous step):

Reviewer 2: ```suggestion
Here are the steps we need to take:

1. Create a `mod1` directory in the `src` crate
2. Create `mod1.cairo` alongside this directory. 
3. Create your source file inside of `mod1` (i.e. `functions.cairo` or any suitable name). Define your code here.
4. Declare the source file/files in `mod1.cairo`. The file contents should look like this (assuming you have `functions.cairo` from the previous step):
```
